File: src/0383-ransom-note/solution.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(ransomNote, magazine) {
    for (var i = 0; i < ransomNote.length; i++) {
        var index = magazine.indexOf(ransomNote[i]);
        if (index === -1) {
            return false;
        }
        magazine = magazine.substring(0, index) + magazine.substring(index + 1);
    }
    return true;
};

File: src/0383-ransom-note/solution.go
package solution

import (
	"strings"
)

func canConstruct(ransomNote string, magazine string) bool {
	for i := 0; i < len(ransomNote); i++ {
		char := string(ransomNote[i])
		if !strings.Contains(magazine, char) {
			return false
		}
		magazine = strings.Replace(magazine, char, "", 1)
	}
	return true
}
